WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift

```diff
@@ -27,7 +28,27 @@ extension Product: InventoryItem {
     func retrieveName(with stores: StoresManager, siteID: Int64) async throws -> String {
         name
     }
+
+    func updateStockQuantity(with newQuantity: Decimal, stores: StoresManager) async throws {
+        return try await withCheckedThrowingContinuation { continuation in
+            let newProduct = copy(stockQuantity: newQuantity)
+
+            let action = ProductAction.updateProduct(product: newProduct) { result in
+                switch result {
+                case .success(_):
+                    continuation.resume(with: .success(()))
+                case let .failure(error):
+                    continuation.resume(throwing: error)
+                }
```

Comment:
```suggestion
                continuation.resume(with: result.map({ _ in () }))
```

Not sure if that's really _better_, because the map is weird, but it's a way we could shorten this if you wanted?

Reply:
That's interesting, is that throwing the specific error when it fails? I don't see how it does it 😅

Reply:
`Result.map` only applies the function to `success` results, there's a corresponding `mapError` function too. The `failures` are unchanged, and the continuation is resumed with the `failure` result, including the associated error.